test/testConfigure.js

```diff
@@ -635,6 +638,128 @@ describe('Configure', function() {
     .then(complete);
   });
 
+  it('does not overwrite "language", "node_js", "install" and "script" in an already existing .travis.yml file', function() {
+    nockBasicPostAuthFlow();
+    configure();
+
+    writeYaml.sync('.travis.yml', {
+      language: 'c',
+      node_js: [ '4.2' ],
+      install: 'run something 1',
+      script: 'run something 2',
+    });
+
+    return enterUsernamePassword()
+    .then(function() {
+      return await('You\'re ready to auto-deploy using Travis!');
+    })
+    .then(function() {
+      var travisYml = readYaml.sync('.travis.yml');
+      expect(travisYml.language).to.equal('c');
+      expect(travisYml.node_js).to.deep.equal(['4.2']);
+      expect(travisYml.install).to.equal('run something 1');
+      expect(travisYml.script).to.equal('run something 2');
+    });
+  });
+
+  it('does not remove environment variables in an already existing .travis.yml file', function() {
+    nockBasicPostAuthFlow();
+    configure();
+
+    writeYaml.sync('.travis.yml', {
+      env: {
+        global: [ 'ENV_GLOBAL' ],
+        matrix: [ 'ENV_MATRIX' ],
+      },
+    });
+
+    return enterUsernamePassword()
+    .then(function() {
+      return await('You\'re ready to auto-deploy using Travis!');
+    })
+    .then(function() {
+      var travisYml = readYaml.sync('.travis.yml');
+      expect(travisYml).to.include.keys('env');
+      expect(travisYml.env).to.include.keys('global');
+      expect(travisYml.env).to.include.keys('matrix');
+      expect(travisYml.env.global).to.have.length(2);
+      expect(travisYml.env.global[0]).to.equal('ENV_GLOBAL');
+      expect(travisYml.env.global[1]).to.have.keys('secure');
```

Comment:
I added this check to make sure we don't overwrite the already existing env in .travis.yml, but we add the one we need.